### PR TITLE
feat(openapi): add openapi gen

### PR DESCRIPTION
**Pull request by liubog2008**
**What this PR does**
- [x] integrate kube-openapi with our Defininition struct
- [x] support our own tag
- [x] example
- [x]  test

**why we need it**

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/openapi-gen/examples/pkg/api/v1/definition.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	},
+}
+
+// Application defines application api model
+type Application struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+	Target    string `json:"target"`
+	Target1   int    `json:"target2"`
+	Target2   bool   `json:"target1"`
```
Comment (Jimexist) [inline cmd/openapi-gen/examples/pkg/api/v1/definition.go]: wrong

Comment (liubog2008) [reply cmd/openapi-gen/examples/pkg/api/v1/definition.go]: Example is copied from @kdada 's example.
It is used to show name in api is generated from json tag.

**Comment by ddysher** [comment]
@liubog2008 filed issue to track unit test.

Is it possible to separate the PR into different small PRs?

**Comment by liubog2008** [comment]
deadline for gometalinter should be added

**Pull request merged**